unified_planning/solvers/factory.py

```diff
@@ -97,20 +101,35 @@ def _get_solver(self, solver_kind: str, name: Optional[str] = None,
             solvers = []
             for name, param in zip(names, params):
                 SolverClass = self._get_solver_class(solver_kind, name)
+                credits = SolverClass.get_credits(**param)
+                if self.env.credits_stream is not None and credits is not None:
+                    self.env.credits_stream.write('You are using ')
+                    credits.write_credits(self.env.credits_stream)
                 solvers.append((SolverClass, param))
-            return up.solvers.parallel.Parallel(solvers)
+            p_solver = up.solvers.parallel.Parallel(solvers)
+            return p_solver
         else:
             if params is None:
                 params = {}
             assert isinstance(params, Dict)
             SolverClass = self._get_solver_class(solver_kind, name, problem_kind, optimality_guarantee)
+            credits = SolverClass.get_credits(**params)
+            if self.env.credits_stream is not None and credits is not None:
+                self.env.credits_stream.write('You are using ')
+                credits.write_credits(self.env.credits_stream)
             return SolverClass(**params)
 
+    @property
+    def env(self) -> 'Environment':
```

Comment:
"Explicit is better than implicit"

```suggestion
    def environment(self) -> 'Environment':
```